app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,501 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {IntlShape, useIntl} from 'react-intl';
+import {Keyboard, View, FlatList} from 'react-native';
+import {ImageResource, Navigation, OptionsTopBarButton} from 'react-native-navigation';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+import {switchToChannel} from '@actions/local/channel';
+import {fetchArchivedChannels, fetchChannels, fetchSharedChannels, joinChannel} from '@actions/remote/channel';
+import useDidUpdate from '@app/hooks/did_update';
+import FormattedText from '@components/formatted_text';
+import Loading from '@components/loading';
+import SearchBar from '@components/search_bar';
+import {General} from '@constants';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {alertErrorWithFallback} from '@utils/draft';
+import {
+    changeOpacity,
+    makeStyleSheetFromTheme,
+    getKeyboardAppearanceFromTheme,
+} from '@utils/theme';
+
+import {dismissModal, goToScreen, setButtons} from '../navigation';
+
+import {ChannelDropdown} from './channel_dropdown';
+import ChannelListRow from './channel_list_row';
+import {ARCHIVED, PUBLIC, SHARED} from './constants';
+
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+
+type Props = {
+
+    // Screen Props (do not change during the lifetime of the screen)
+    componentId: string;
+    categoryId?: string;
+    closeButton: ImageResource;
+
+    // Calculated Props
+    canCreateChannels: boolean;
+    currentUserId: string;
+    currentTeamId: string;
+    joinedChannels?: MyChannelModel[];
+    sharedChannelsEnabled?: boolean;
+    canShowArchivedChannels: boolean;
+}
+
+const filterChannelsByTerm = (channels: Channel[], term: string) => {
+    const lowerCasedTerm = term.toLowerCase();
+    return channels.filter((c) => {
+        return (c.name.toLowerCase().includes(lowerCasedTerm) || c.display_name.toLowerCase().includes(lowerCasedTerm));
+    });
+};
+
+const filterJoinedChannels = (joinedChannels: MyChannelModel[], allChannels: Channel[] | undefined) => {
+    const ids = joinedChannels.map((c) => c.id);
+    return allChannels?.filter((c) => !ids.includes(c.id));
+};
+
+const makeLeftButton = (icon: ImageResource): OptionsTopBarButton => {
+    return {
+        id: 'close-more-channels',
+        icon,
+        testID: 'close.more_channels.button',
```

Comment:
There are many testIDs and intl ids that reference more_channels instead of browse_channels. Should we maintain them, or should we change them now? (we may have to change also all the i18n files to keep the translations)

Reply:
we can change them, reality is that in v1 this screen was called `more_channels`. If we change it, it should be only in the code and in `en.json` forget about the rest, translation server will do it's job and help translators take care of this.